beamer/agent/chain.py

```diff
@@ -611,12 +615,7 @@ def maybe_prove(claim: Claim, context: Context) -> None:
     if not can_prove:
         return
 
-    if request.fill_tx is not None:
-        proof_tx = request.fill_tx
-    elif claim.invalidation_tx is not None:
-        proof_tx = claim.invalidation_tx
-
-    if proof_tx in context.l1_resolutions:
+    if proved_tx is None or proved_tx in context.l1_resolutions:
```

Comment:
There is an additional check missing here, that if claim.proved_tx is not None then it should exit. 
Note, in this part we are sure that we are an OP chain. If one of the claim.proved_tx is already proved, this is the one which should be resolved in the future.